src/lib/Sensor.js

```diff
@@ -11,14 +11,12 @@ class Sensor extends EventEmitter {
     this.onlyIfChanged = onlyIfChanged;
 
     if (enablePolling) {
-      setInterval(() => {
-        const newTemp = this.getTemperature();
-
+      setInterval(() => this.getTemperatureAsync().catch(() => false).then(newTemp => {
```

Comment:
There is `.catch()` method which returns `false` in case of error to keep 100% backward compatibility with solution based on synchronous function which returns `false` in case of an error.